Adding k-diagram recipe

earthai-tech opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

earthai-tech committed:
Add k-diagram receipe

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/k-diagram/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/k-diagram/meta.yaml**:

 * ❌ license_file entry is missing, but is required.

For **recipes/k-diagram/meta.yaml**:

 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18150859555](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18150859555). Examine the logs at this URL for more detail.</sub>

earthai-tech committed:
Add license

earthai-tech committed:
Update pypi.org instead of pypi.io

earthai-tech committed:
uploaded normalized package name

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/k-diagram/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/k-diagram/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18152989971](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18152989971). Examine the logs at this URL for more detail.</sub>

earthai-tech committed:
set python_min

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/k-diagram/meta.yaml```) and found it was in an excellent condition.

earthai-tech committed:
update meta.yml python min

earthai-tech committed:
follow noarch doc

earthai-tech committed:
add type-extensions

earthai-tech committed:
Update meta.yaml

Co-authored-by: Travis Hathaway <travis.j.hathaway@gmail.com>

earthai-tech commented:
> Hi @earthai-tech,
> 
> This recipe looks like it's reasonable shape. I do have a couple suggestions though. I've also fixed some broken links. Please let us know if you're still interested in going forward. We apologize for the long wait time for a review.

Sure , I'm still interesting for going forward. I Know you're too busy so there is no problem for that.

earthai-tech committed:
Update meta.yaml

earthai-tech commented:
> Hi @earthai-tech,
> 
> Happy to hear you're still willing to add this. Can you please remove the `LICENSE` like I asked and also fix the broken link in the recipe?

Thanks Travis. I removed the license_file entry and fixed the broken GitHub links from earth-ai-tech to earthai-tech. The PR has been updated.

earthai-tech committed:
Merge branch 'main' into main

earthai-tech committed:
Delete recipes/k-diagram/LICENSE

earthai-tech committed:
Update meta.yaml

earthai-tech commented on recipes/k-diagram/LICENSE:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
> @earthai-tech,
> 
> Sorry for the confusion 🙈. I meant to delete this file. This license file is already present in your repository (https://github.com/earthai-tech/k-diagram), and is included in the tar.gz file you have uploaded to PyPI, so we don't need it here.

Thanks Travis, sorry for the misunderstanding. I have now deleted the extra `recipes/k-diagram/LICENSE` file from the recipe and committed the change.

I also restored `license_file: LICENSE` in `meta.yaml`, so the recipe now points to the LICENSE included in the PyPI source tarball.

Please let me know if anything else needs to be adjusted.

earthai-tech committed:
Merge branch 'main' into main

earthai-tech committed:
Merge branch 'main' into main

isuruf committed:
Merge branch 'main' into main
